cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -0,0 +1,184 @@
+                sw360ProjectClient.addSW360ReleasesToSW360Project(it.id, linkedReleases)
+            }
+        }
+    }
+
+    private fun createSw360Project(project: Project, client: SW360ProjectClientAdapter): SW360Project? {
+        val sw360Project = SW360Project()
+            .setName(project.id.name)
+            .setVersion(project.id.version)
+            .setDescription("A ${project.id.type} project with the PURL ${project.id.toPurl()}.")
+            .setVisibility(SW360Visibility.BUISNESSUNIT_AND_MODERATORS)
+
+        return try {
+            val createdProject = client.createProject(sw360Project)
+            createdProject?.let {
+                log.debug { "Project ${createdProject.name}-${createdProject.version} created in SW360." }
+            }
+
+            createdProject
+        } catch (e: SW360ClientException) {
+            log.error {
+                "Could not create the project ${project.id.name}-${project.id.version} in SW360: " +
+                        e.collectMessagesAsString()
+            }
+
+            null
+        }
+    }
+
+    private fun createSw360Release(pkg: Package, client: SW360ReleaseClientAdapter): SW360Release? {
+        val sw360Release = SW360Release()
+            .setMainLicenseIds(pkg.declaredLicenses)
+            .setName(pkg.id.name)
+            .setVersion(pkg.id.version)
+
+        return try {
+            val createRelease = client.createRelease(sw360Release)
+            createRelease?.let {
+                log.debug { "Release ${createRelease.name}-${createRelease.version} created in SW360." }
+            }
+
+            createRelease
+        } catch (e: SW360ClientException) {
+            log.error {
+                "Could not create the release for ${pkg.id.toCoordinates()} in SW360: " + e.collectMessagesAsString()
+            }
+
+            null
+        }
+    }
+
+    private fun createSw360Connection(config: Sw360StorageConfiguration, jsonMapper: ObjectMapper): SW360Connection {
+        val httpClientConfig: HttpClientConfig = HttpClientConfig
+            .basicConfig()
+            .withObjectMapper(jsonMapper)
+        val httpClient: HttpClient = HttpClientFactoryImpl().newHttpClient(httpClientConfig)
+
+        val sw360ClientConfig: SW360ClientConfig = SW360ClientConfig.createConfig(
+            config.restUrl,
+            config.authUrl,
+            config.username,
+            config.password,
+            config.clientId,
+            config.clientPassword,
+            httpClient,
+            jsonMapper
+        )
+
+        return SW360ConnectionFactory().newConnection(sw360ClientConfig)
+    }
+
+    private fun getProjectWithPackages(ortResult: OrtResult): Map<Project, List<Package>> =
+        ortResult.getProjects(omitExcluded = true).map { project ->
+            project to ortResult.getPackages().filter { (pkg, _) -> pkg.id in project.collectDependencies() }
+                .map { (pkg, _) -> pkg }
+        }.toMap()
```

Comment:
I believe this could be replaced with

        ortResult.getProjects(omitExcluded = true).associateWith { project ->
            project.collectDependencies().mapNotNull { ortResult.getUncuratedPackageById(it) }
        }